### PR TITLE
Add "getDynamicValue" for SizeAdd node.

### PR DESCRIPTION
**Pull request by vanbasten23**

### CODE REVIEW

**Comment by miladm** [comment]
Perhaps you want to rebase `master` to get the latest of `SizeNode` class changes?

**Comment by JackCaoG** [comment]
merge doesn't really work given how CI is setup, unless you squsah all commit into one. Otherwise use rebase instead

File: test/cpp/test_ir.cpp [hunk 1]
```diff
@@ -181,6 +182,31 @@ TEST_F(IrTest, TestSizeAddNode) {
   });
 }
 
+TEST_F(IrTest, TestSizeAddNodeDynamic) {
```
Comment (Krovatkin) [inline test/cpp/test_ir.cpp:185]: nit: maybe, 
* add a recursive `(a.size(0) + a.size(1)) + a.size(0)` ?
* use different tensors `a.size(0) + b.size(1)`

I wouldn't push for these tests though

**Review by Krovatkin — APPROVED** [review]
:shipit:

Would be curious how we are going to do `SizeAdd::Lower` **wink** **wink** @JackCaoG

**Review by miladm — APPROVED** [review]
Thanks.

**Comment by vanbasten23** [comment]
> :shipit:
> 
> Would be curious how we are going to do `SizeAdd::Lower` **wink** **wink** @JackCaoG

It seems SizeAdd::Lower has already been implemented.

**Comment by vanbasten23** [comment]
> Perhaps you want to rebase `master` to get the latest of `SizeNode` class changes?

Done.

File: test/cpp/test_ir.cpp (hunk 1 above)
Comment (vanbasten23) [reply test/cpp/test_ir.cpp]: Added a test for using different tensors.

**Comment by vanbasten23** [comment]
> merge doesn't really work given how CI is setup, unless you squsah all commit into one. Otherwise use rebase instead

Rebased.

**Review by JackCaoG — APPROVED** [review]
Thanks!

**Review by miladm — COMMENTED** [review]
Reviewed this PR. LGTM. Thanks @vanbasten23!

**Pull request merged**